lib/src/components/NxCombobox/NxCombobox.tsx

```diff
@@ -0,0 +1,318 @@
+      }
+      setFocusableBtnIndex(null);
+      setShowDropdown(false);
+    }
+  }
+
+  function handleOnChange(newVal: string) {
+    setFocusableBtnIndex(null);
+    onChange(newVal);
+
+    if (newVal && newVal.trim() !== value.trim()) {
+      doSearch(newVal);
+    }
+    else if (!newVal) {
+      setShowDropdown(false);
+    }
+  }
+
+  function doSearch(val: string) {
+    focusTextInput();
+    onSearch(val.trim());
+    setShowDropdown(true);
+  }
+
+  function focusTextInput() {
+    inputRef.current?.querySelector('input')?.focus();
+  }
+
+  // helper for focusing different buttons in the dropdown menu
+  const adjustBtnFocus = (adjust: (i: number) => number) => () => {
+        const newFocusableBtnIndex = adjust(focusableBtnIndex ?? 0),
+            elToFocus = dropdownRef.current?.children[newFocusableBtnIndex] as HTMLElement | null;
+
+        if (elToFocus) {
+          setFocusableBtnIndex(newFocusableBtnIndex);
+          elToFocus.scrollIntoView({ block: 'nearest' });
+
+          if (autoComplete) {
+            setInlineStyle(false);
+          }
+        }
+      },
+      focusNext = adjustBtnFocus(inc),
+      focusPrev = adjustBtnFocus(dec),
+      focusFirst = adjustBtnFocus(always(0)),
+      focusLast = adjustBtnFocus(always(matches.length - 1));
+
+  function handleOnClick({displayName}: DataItem<T, string>) {
+    onChange(displayName);
+    onSearch(displayName);
+    focusTextInput();
+    setFocusableBtnIndex(null);
+    setShowDropdown(false);
+  }
+
+  function handleKeyDown(evt: KeyboardEvent<HTMLElement>) {
+    const inputEle = evt.currentTarget as HTMLInputElement,
+        elToFocusText = focusableBtnIndex !== null ? matches[focusableBtnIndex].displayName : null;
+    switch (evt.key) {
+      case 'Enter':
+        if (focusableBtnIndex !== null) {
+          handleOnClick(matches[focusableBtnIndex]);
+        }
+        evt.preventDefault();
+        break;
+      case 'Home':
+        inputEle.setSelectionRange(0, 0);
+        evt.preventDefault();
+        break;
+      case 'End':
+        autoComplete && elToFocusText
+          ? inputEle.setSelectionRange(elToFocusText.length, elToFocusText.length)
+          : inputEle.setSelectionRange(value.length, value.length);
+        evt.preventDefault();
+        break;
+      case 'ArrowDown':
+        if (!isEmpty) {
+          setShowDropdown(true);
+        }
+
+        focusableBtnIndex !== null
+          ? (focusableBtnIndex === matches.length - 1 ? focusFirst() : focusNext())
+          : focusFirst();
+
+        evt.preventDefault();
+        break;
+      case 'ArrowUp':
+        if (!isEmpty) {
+          setShowDropdown(true);
+        }
+
+        focusableBtnIndex !== null
+          ? (focusableBtnIndex === 0 ? focusLast() : focusPrev())
+          : focusLast();
+
+        evt.preventDefault();
+        break;
+      case 'Backspace':
+        if (autoComplete) {
+          setInlineStyle(false);
+        }
```

Comment:
No real need to check this condition
```suggestion
        setInlineStyle(false);
```